File: TOMIMSV4/Entity/Transacciones/OrdenCompra/OC_Encabezado/clsBeTrans_oc_enc.vb
Public Class clsBeTrans_oc_enc
    Implements ICloneable
    Implements IDisposable

    Public Property IdOrdenCompraEnc() As Integer = 0
    Public Property IdPropietarioBodega() As Integer = 0
    Public Property IdProveedorBodega() As Integer = 0
    Public Property IdTipoIngresoOC() As Integer = 0
    Public Property IdEstadoOC() As Integer = 0
    Public Property IdMotivoDevolucion() As Integer = 0
    Public Property Fecha_Creacion() As Date = New Date(1900, 1, 1)
    Public Property Hora_Creacion() As Date = New Date(1900, 1, 1)
    ''' <summary>
    ''' #EJC20240606: DocNum SAP.
    ''' </summary>
    ''' <returns></returns>
    Public Property No_Documento() As String = ""
    Public Property User_Agr() As String = ""
    Public Property Fec_Agr() As Date = Now
    Public Property User_Mod() As String = ""
    Public Property Fec_Mod() As Date = New Date(1900, 1, 1)
    Public Property Procedencia() As String = ""
    Public Property No_Marchamo() As String = ""
    Public Property Referencia() As String = ""
    Public Property Observacion() As String = ""
    Public Property Control_Poliza() As Boolean = False
    Public Property Activo() As Boolean = False
    Public Property Fecha_Recepcion() As Date = New Date(1900, 1, 1)
    Public Property Hora_Inicio_Recepcion() As Date = New Date(1900, 1, 1)
    Public Property Hora_Fin_Recepcion() As Date = New Date(1900, 1, 1)
    Public Property IdMuelleRecepcion() As Integer = 0
    Public Property Programar_Recepcion() As Boolean = False
    Public Property IdMotivoAnulacionBodega() As Integer = 0
    Public Property Enviado_A_ERP As Boolean = False
    Public Property No_Ticket_TMS As String = ""
    Public Property IdNoDocumentoRef As Integer = 0

    '#EJC20190314_0701: Control de series y correlativos por documento en O.C.
    Public Property Serie As String = ""
    Public Property Correlativo As String = ""

    Public Property IdOperadorBodegaDefecto As Integer = 0

    ''' <summary>
    ''' Se utiliza para la gestión del acuerdo comercial relacionado a un proveedor en CEALSA.
    ''' </summary>
    ''' <returns></returns>
    Public Property IdAcuerdoComercial As Integer = 0

    ''' <summary>
    ''' Si es una transferencia de otra bodega de WMS aquí estará contenido el IdDespachoEnc de la bodega origen.
    ''' </summary>
    ''' <returns></returns>
    Public Property IdDespachoEnc As Integer = 0

    ''' <summary>
    ''' #EJC20210427: Se utiliza para almacenar el número de documento de recepción del mdulo de bodega de NAV para la bodega de PT.
    ''' </summary>
    ''' <returns></returns>
    Public Property No_Documento_Recepcion_ERP As String = ""
    Public Property No_Documento_Devolucion() As String = ""
    Public Property IdPedidoEncDevolucion() As Integer = 0
    Public Property Push_To_NAV As Boolean = False
    Public Property No_Documento_Ubicacion_ERP As String = ""
    Public Property PutAway_Registrado As Boolean = False
    Public Property Codigo_Empresa_ERP As String = ""
    Public Property IdCampaña As Integer = 0
    Sub New()
    End Sub

    Sub New(ByRef IdOrdenCompraEnc As Integer, ByVal IdPropietarioBodega As Integer, ByVal IdProveedorBodega As Integer, ByVal IdTipoIngresoOC As Integer, ByVal IdEstadoOC As Integer, ByVal IdMotivoDevolucion As Integer, ByVal Fecha_Creacion As Date, ByVal Hora_Creacion As Date, ByVal No_Documento As String, ByVal User_Agr As String, ByVal Fec_Agr As Date, ByVal User_Mod As String, ByVal Fec_Mod As Date, ByVal Procedencia As String, ByVal No_Marchamo As String, ByVal Referencia As String, ByVal Observacion As String, ByVal Control_Poliza As Boolean, ByVal Activo As Boolean, ByVal Fecha_Recepcion As Date, ByVal Hora_Inicio_Recepcion As Date, ByVal Hora_Fin_Recepcion As Date, ByVal IdMuelleRecepcion As Integer, ByVal Programar_Recepcion As Boolean, ByVal IdMotivoAnulacionBodega As Integer)
        Me.IdOrdenCompraEnc = IdOrdenCompraEnc
        Me.IdPropietarioBodega = IdPropietarioBodega
        Me.IdProveedorBodega = IdProveedorBodega
        Me.IdTipoIngresoOC = IdTipoIngresoOC
        Me.IdEstadoOC = IdEstadoOC
        Me.IdMotivoDevolucion = IdMotivoDevolucion
        Me.Fecha_Creacion = Fecha_Creacion
        Me.Hora_Creacion = Hora_Creacion
        Me.No_Documento = No_Documento
        Me.User_Agr = User_Agr
        Me.Fec_Agr = Fec_Agr
        Me.User_Mod = User_Mod
        Me.Fec_Mod = Fec_Mod
        Me.Procedencia = Procedencia
        Me.No_Marchamo = No_Marchamo
        Me.Referencia = Referencia
        Me.Observacion = Observacion
        Me.Control_Poliza = Control_Poliza
        Me.Activo = Activo
        Me.Fecha_Recepcion = Fecha_Recepcion
        Me.Hora_Inicio_Recepcion = Hora_Inicio_Recepcion
        Me.Hora_Fin_Recepcion = Hora_Fin_Recepcion
        Me.IdMuelleRecepcion = IdMuelleRecepcion
        Me.Programar_Recepcion = Programar_Recepcion
        Me.IdMotivoAnulacionBodega = IdMotivoAnulacionBodega
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/Trans_oc/clsBeTrans_oc_enc.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;
using WMS.EntityCore.I_nav_Ped_Compra;
using WMS.EntityCore.Propietario;
using WMS.EntityCore.Proveedor;
using WMSWebAPI.Be;

namespace WMS.EntityCore.Trans_oc
{
    public class clsBeTrans_oc_enc : ICloneable
    {
        [Column("IdOrdenCompraEnc")]
        [DisplayName("IdOrdenCompraEnc")]
        public int IdOrdenCompraEnc { get; set; } = 0;

        [Column("IdPropietarioBodega")]
        [DisplayName("IdPropietarioBodega")]
        public int IdPropietarioBodega { get; set; } = 0;

        [Column("IdProveedorBodega")]
        [DisplayName("IdProveedorBodega")]
        public int IdProveedorBodega { get; set; } = 0;

        [Column("IdTipoIngresoOC")]
        [DisplayName("IdTipoIngresoOC")]
        public int IdTipoIngresoOC { get; set; } = 0;

        [Column("IdEstadoOC")]
        [DisplayName("IdEstadoOC")]
        public int IdEstadoOC { get; set; } = 0;

        [Column("IdMotivoDevolucion")]
        [DisplayName("IdMotivoDevolucion")]
        public int IdMotivoDevolucion { get; set; } = 0;

        [Column("Fecha_Creacion")]
        [DisplayName("Fecha_Creacion")]
        public DateTime Fecha_Creacion { get; set; } = DateTime.Now;

        [Column("Hora_Creacion")]
        [DisplayName("Hora_Creacion")]
        public DateTime Hora_Creacion { get; set; } = DateTime.Now;

        [Column("No_Documento")]
        [DisplayName("No_Documento")]
        public string No_Documento { get; set; } = "";

        [Column("User_Agr")]
        [DisplayName("User_Agr")]
        public string User_Agr { get; set; } = "";

        [Column("Fec_Agr")]
        [DisplayName("Fec_Agr")]
        public DateTime Fec_Agr { get; set; } = DateTime.Now;

        [Column("User_Mod")]
        [DisplayName("User_Mod")]
        public string User_Mod { get; set; } = "";

        [Column("Fec_Mod")]
        [DisplayName("Fec_Mod")]
        public DateTime Fec_Mod { get; set; } = DateTime.Now;

        [Column("Procedencia")]
        [DisplayName("Procedencia")]
        public string Procedencia { get; set; } = "";

        [Column("No_Marchamo")]
        [DisplayName("No_Marchamo")]
        public string No_Marchamo { get; set; } = "";

        [Column("Referencia")]
        [DisplayName("Referencia")]
        public string Referencia { get; set; } = "";

        [Column("Observacion")]
        [DisplayName("Observacion")]
        public string Observacion { get; set; } = "";

        [Column("Control_Poliza")]
        [DisplayName("Control_Poliza")]
        public bool Control_Poliza { get; set; } = false;

        [Column("Activo")]
        [DisplayName("Activo")]
        public bool Activo { get; set; } = false;

        [Column("Fecha_Recepcion")]
        [DisplayName("Fecha_Recepcion")]
        public DateTime Fecha_Recepcion { get; set; } = DateTime.Now;

        [Column("Hora_Inicio_Recepcion")]
        [DisplayName("Hora_Inicio_Recepcion")]
        public DateTime Hora_Inicio_Recepcion { get; set; } = DateTime.Now;

        [Column("Hora_Fin_Recepcion")]
        [DisplayName("Hora_Fin_Recepcion")]
        public DateTime Hora_Fin_Recepcion { get; set; } = DateTime.Now;

        [Column("IdMuelleRecepcion")]
        [DisplayName("IdMuelleRecepcion")]
        public int IdMuelleRecepcion { get; set; } = 0;

        [Column("Programar_Recepcion")]
        [DisplayName("Programar_Recepcion")]
        public bool Programar_Recepcion { get; set; } = false;

        [Column("IdMotivoAnulacionBodega")]
        [DisplayName("IdMotivoAnulacionBodega")]
        public int IdMotivoAnulacionBodega { get; set; } = 0;

        [Column("Enviado_A_ERP")]
        [DisplayName("Enviado_A_ERP")]
        public bool Enviado_A_ERP { get; set; } = false;

        [Column("serie")]
        [DisplayName("serie")]
        public string Serie { get; set; } = "";

        [Column("correlativo")]
        [DisplayName("correlativo")]
        public int Correlativo { get; set; } = 0;

        [Column("IdDespachoEnc")]
        [DisplayName("IdDespachoEnc")]
        public int IdDespachoEnc { get; set; } = 0;

        [Column("no_ticket_tms")]
        [DisplayName("no_ticket_tms")]
        public string No_ticket_tms { get; set; } = "";

        [Column("IdNoDocumentoRef")]
        [DisplayName("IdNoDocumentoRef")]
        public int IdNoDocumentoRef { get; set; } = 0;

        [Column("idacuerdocomercial")]
        [DisplayName("idacuerdocomercial")]
        public int Idacuerdocomercial { get; set; } = 0;

        [Column("IdOperadorBodegaDefecto")]
        [DisplayName("IdOperadorBodegaDefecto")]
        public int IdOperadorBodegaDefecto { get; set; } = 0;

        [Column("IdBodega")]
        [DisplayName("IdBodega")]
        public int IdBodega { get; set; } = 0;

        [Column("no_documento_recepcion_erp")]
        [DisplayName("no_documento_recepcion_erp")]
        public string No_documento_recepcion_erp { get; set; } = "";

        [Column("no_documento_devolucion")]
        [DisplayName("no_documento_devolucion")]
        public string No_documento_devolucion { get; set; } = "";

        [Column("IdPedidoEncDevolucion")]
        [DisplayName("IdPedidoEncDevolucion")]
        public int IdPedidoEncDevolucion { get; set; } = 0;

        [Column("push_to_nav")]
        [DisplayName("push_to_nav")]
        public bool Push_to_nav { get; set; } = false;

        [Column("no_documento_ubicacion_erp")]
        [DisplayName("no_documento_ubicacion_erp")]
        public string No_documento_ubicacion_erp { get; set; } = "";

        [Column("PutAway_Registrado")]
        [DisplayName("PutAway_Registrado")]
        public bool PutAway_Registrado { get; set; } = false;

        [Column("Codigo_Empresa_ERP")]
        [DisplayName("Codigo_Empresa_ERP")]
        public string Codigo_Empresa_ERP { get; set; } = "";

        public int IdCampaña { get; set; } = 0;
        public string Usr_Documento { get; set; } = "";
        public string Comentarios { get; set; } = "";
        public List<clsBeTrans_oc_det> DetalleOC { get; set; } = new List<clsBeTrans_oc_det>();
        public List<clsBeTrans_oc_det_lote> DetalleLotes { get; set; } = new List<clsBeTrans_oc_det_lote>();
        public List<clsBeI_nav_barras_pallet> DetallePallets { get; set; } = new List<clsBeI_nav_barras_pallet>();
        public clsBeTrans_oc_pol? ObjPoliza { get; set; } = new clsBeTrans_oc_pol();
        public List<clsBeTrans_oc_imagen> ListaImg { get; set; } = new List<clsBeTrans_oc_imagen>();
        public clsBePropietario_bodega PropietarioBodega { get; set; } = new clsBePropietario_bodega();
        public clsBeProveedor_bodega ProveedorBodega { get; set; } = new clsBeProveedor_bodega();
        public clsBeTrans_oc_estado? EstadoOC { get; set; } = new clsBeTrans_oc_estado();
        public bool IsNew { get; set; }
        public bool EsDevolucion { get; set; }
        public clsBeTrans_oc_ti? TipoIngreso { get; set; } = new clsBeTrans_oc_ti();
        public bool ExisteRecepcionNoFinalizada { get; set; }
        public string No_Documento_Recepcion_ERP { get; set; } = "";

        public clsBeTrans_oc_enc() { }

        public object Clone()
        {
            return MemberwiseClone();
        }
    }
}